odoo/addons/base/models/res_partner.py

```diff
@@ -696,6 +698,11 @@ def _fields_sync(self, values):
         # 2. To DOWNSTREAM: sync children
         self._children_sync(values)
 
+        # 3. To UPSTREAM: force parent update
```

Comment:
The doc clearly states "sync commercial fields and address fields from company, to children" - not the other way around. Maybe that logic should be in the write method?

Note that commercial fields are marked as readonly when you have a parent and are of type contact.